aptos-move/e2e-move-tests/src/tests/scripts.rs

```diff
@@ -80,3 +81,181 @@ fn read_coin(h: &MoveHarness, account: &AccountAddress) -> u64 {
         .unwrap()
         .coin()
 }
+
+#[test]
+fn test_two_to_two_transfer_gas_payer() {
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::GAS_PAYER_ENABLED], vec![]);
+
+    let alice = h.new_account_at(AccountAddress::from_hex_literal("0xa11ce").unwrap());
+    let bob = h.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
+    let carol = h.new_account_at(AccountAddress::from_hex_literal("0xca501").unwrap());
+    let david = h.new_account_at(AccountAddress::from_hex_literal("0xda51d").unwrap());
+    let payer = h.new_account_at(AccountAddress::from_hex_literal("0xea51d").unwrap());
+
+    let amount_alice = 100;
+    let amount_bob = 200;
+    let amount_carol = 50;
+    let amount_david = amount_alice + amount_bob - amount_carol;
+
+    let build_options = aptos_framework::BuildOptions {
+        with_srcs: false,
+        with_abis: false,
+        with_source_maps: false,
+        with_error_map: false,
+        ..aptos_framework::BuildOptions::default()
+    };
+
+    let package = aptos_framework::BuiltPackage::build(
+        common::test_dir_path("../../../move-examples/scripts/two_by_two_transfer"),
+        build_options,
+    )
+    .expect("building package must succeed");
+
+    let alice_start = read_coin(&h, alice.address());
+    let bob_start = read_coin(&h, bob.address());
+    let carol_start = read_coin(&h, carol.address());
+    let david_start = read_coin(&h, david.address());
+    let payer_start = read_coin(&h, payer.address());
+
+    let code = package.extract_script_code()[0].clone();
+    let script = Script::new(code, vec![], vec![
+        TransactionArgument::U64(amount_alice),
+        TransactionArgument::U64(amount_bob),
+        TransactionArgument::Address(*carol.address()),
+        TransactionArgument::Address(*david.address()),
+        TransactionArgument::U64(amount_carol),
+    ]);
+
+    let transaction = TransactionBuilder::new(alice.clone())
+        .secondary_signers(vec![bob.clone(), payer.clone()])
```

Comment:
Let's add a quick case in this same test for when the secondary signers list only contains the gas payer (effectively a normal tx with a single sender but separate gas payer)

Reply:
I added normal entry function for this